modules/motus/downloaddb/main.nf

```diff
@@ -0,0 +1,39 @@
+process MOTUS_DOWNLOADDB {
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::motus=3.0.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/motus:3.0.1--pyhdfd78af_0':
+        'quay.io/biocontainers/motus:3.0.1--pyhdfd78af_0' }"
+
+    input:
+    path motus_downloaddb
+
+    output:
+    path "db_mOTU"                , emit: db
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    ## must copy file to working directory,
+    ## otherwise the reference_db will be download to bin folder
+    ## other than current directory
+    cp $motus_downloaddb ${motus_downloaddb.simpleName}_copy.py
+    python ${motus_downloaddb.simpleName}_copy.py \\
+        -t $task.cpus
+    ## clean up
+    rm ${motus_downloaddb.simpleName}_copy.py
+
+    ## mOTUs version number is not available from command line.
+    ## mOTUs save the version number in index database folder.
+    ## mOTUs will check the database version is same version as exec version.
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        mOTUs: \$(grep motus db_mOTU/db_mOTU_versions | sed 's/motus\\t//g')
```

Comment:
This seems a bit awkward - I think it's better to do this as done with the HOMER module [^1]

[^1]: https://nf-co.re/docs/contributing/modules#general point 4

Reply:
I still want to keep the version detection via database because the version information is detectable from the database. If the database version is different from software expectation, the tool will through an error.

Reply:
Ok fair enough!